src/app/core/services/filters.service.ts

```diff
@@ -61,21 +64,77 @@ export class FiltersService {
   // Helps in determining whether all milestones were selected from previous repo during sanitization of milestones
   private previousMilestonesLength = 0;
 
-  constructor(private milestoneService: MilestoneService) {}
+  constructor(
+    private logger: LoggingService,
+    private router: Router,
+    private route: ActivatedRoute,
+    private milestoneService: MilestoneService
+  ) {}
+
+  private pushFiltersToUrl(): void {
+    const queryParams = {};
+    for (const filterName of Object.keys(this.filter$.value)) {
+      if (this.filter$.value[filterName] instanceof Set) {
+        queryParams[filterName] = JSON.stringify([...this.filter$.value[filterName]]);
+      } else {
+        queryParams[filterName] = JSON.stringify(this.filter$.value[filterName]);
+      }
+    }
+    queryParams[FiltersService.PRESET_VIEW_QUERY_PARAM_KEY] = this.presetView$.value;
+
+    this.router.navigate([], {
+      relativeTo: this.route,
+      queryParams,
+      queryParamsHandling: 'merge',
+      replaceUrl: true
+    });
+  }
 
   clearFilters(): void {
     this.filter$.next(this.defaultFilter());
     this.presetView$.next('currentlyActive');
     this.previousMilestonesLength = 0;
   }
 
+  initializeFromURLParams() {
+    const nextFilter: Filter = this.defaultFilter();
+    const queryParams = this.route.snapshot.queryParamMap;
+    try {
+      const presetView = queryParams.get(FiltersService.PRESET_VIEW_QUERY_PARAM_KEY);
+
+      // Use preset view if set in url
+      if (presetView !== 'custom') {
+        this.updatePresetView(presetView);
+        return;
+      }
```

Comment:
`null` value and invalid `presetView` are not handled. We can simply handle them by calling `updatePresetView` with the default value.

Reply:
Agreed. I have made the appropriate changes